mapie/regression/utils/ccp_phi_function.py

```diff
@@ -0,0 +1,743 @@
+from __future__ import annotations
+
+import inspect
+from typing import Callable, Dict, List, Optional, Tuple, Union, cast
+import numbers
+import warnings
+
+import numpy as np
+from mapie._typing import NDArray
+from sklearn.utils import _safe_indexing
+from sklearn.utils.validation import _num_samples
+
+
+class PhiFunction():
```

Comment:
Consider to adapt this class to an abstract class. Give us your insight for this proposition.

Reply:
I changed most of the ``PhiFunction`` and its subclasses after your suggestion.
It is now an abstract class on the template ``fit``/``transform``. Indeed ``phi`` is a transformation of ``X``, so it makes sense. Some transformations don't need to be fitted, other require a fit step: 
 - it is the case of the ``GaussianPhiFunction`` class. 
 - It will also be the case if I want to add a ``QuantilePhiFunction`` (a bigger name must be found) which automatically compute groups with X<Q1, Q1<X<Q2, Q2<X<Q3, Q3<X, to have a homogenous prediction interval)